soc/xtensa/dc233c/include/xtensa-dc233c.ld

```diff
@@ -0,0 +1,501 @@
+    *(.literal .text .literal.* .text.* .stub .gnu.warning .gnu.linkonce.literal.* .gnu.linkonce.t.*.literal .gnu.linkonce.t.*)
+    *(.fini.literal)
+    KEEP(*(.fini))
+    *(.gnu.version)
+
+    #include <zephyr/linker/kobject-text.ld>
+
+    MMU_PAGE_ALIGN
+
+    _text_end = ABSOLUTE(.);
+    _etext = .;
+  } >RAM :sram0_phdr
+  __text_region_end = .;
+
+  .rodata : HDR_MMU_PAGE_ALIGN
+  {
+    __rodata_region_start = ABSOLUTE(.);
+    *(.rodata)
+    *(.rodata.*)
+    *(.gnu.linkonce.r.*)
+    *(.rodata1)
+
+    . = ALIGN(4);
+    #include <snippets-rodata.ld>
+    #include <zephyr/linker/kobject-rom.ld>
+
+    . = ALIGN(4);		/* this table MUST be 4-byte aligned */
+    _bss_table_start = ABSOLUTE(.);
+    LONG(_bss_start)
+    LONG(_bss_end)
+    _bss_table_end = ABSOLUTE(.);
+
+    MMU_PAGE_ALIGN
+
+    __rodata_region_end = ABSOLUTE(.);
+  } >RAM :sram0_phdr
+
+#include <zephyr/linker/common-rom.ld>
+
+#include <zephyr/linker/cplusplus-rom.ld>
+
+#include <snippets-sections.ld>
+
+  .data : HDR_MMU_PAGE_ALIGN
+  {
+    _image_ram_start = ABSOLUTE(.);
+    __data_start = ABSOLUTE(.);
+    *(.data)
+    *(.data.*)
+    *(.gnu.linkonce.d.*)
+    KEEP(*(.gnu.linkonce.d.*personality*))
+    *(.data1)
+    *(.sdata)
+    *(.sdata.*)
+    *(.gnu.linkonce.s.*)
+    *(.sdata2)
+    *(.sdata2.*)
+    *(.gnu.linkonce.s2.*)
+    KEEP(*(.jcr))
+
+    . = ALIGN(4);
+    #include <snippets-rwdata.ld>
+    . = ALIGN(4);
+
+    MMU_PAGE_ALIGN
+
+    __data_end = ABSOLUTE(.);
+  } >RAM :sram0_phdr
+
+#include <snippets-data-sections.ld>
+
+#include <zephyr/linker/common-ram.ld>
+
+#include <zephyr/linker/cplusplus-ram.ld>
+
+#include <snippets-ram-sections.ld>
+
+  .bss (NOLOAD) : HDR_MMU_PAGE_ALIGN
+  {
+    . = ALIGN (8);
+    _bss_start = ABSOLUTE(.);
+    *(.dynsbss)
+    *(.sbss)
+    *(.sbss.*)
+    *(.gnu.linkonce.sb.*)
+    *(.scommon)
+    *(.sbss2)
+    *(.sbss2.*)
+    *(.gnu.linkonce.sb2.*)
+    *(.dynbss)
+    *(.bss)
+    *(.bss.*)
+    *(.gnu.linkonce.b.*)
+    *(COMMON)
+    *(.sram.bss)
+    . = ALIGN (8);
+    _bss_end = ABSOLUTE(.);
+
+    MMU_PAGE_ALIGN
+
+  } >RAM :sram0_bss_phdr
+
+#include <zephyr/linker/common-noinit.ld>
+
+/* Must be last in RAM */
+#include <zephyr/linker/kobject-data.ld>
+
+#include <zephyr/linker/ram-end.ld>
+
+  _heap_start = .;
+
+  PROVIDE(_heap_sentry = ORIGIN(RAM) + LENGTH(RAM));
+  PROVIDE(_heap_end = ORIGIN(RAM) + LENGTH(RAM));
+
+  PROVIDE(__stack = z_interrupt_stacks + CONFIG_ISR_STACK_SIZE);
+
+#include <zephyr/linker/debug-sections.ld>
+
+  .xtensa.info  0 :  { *(.xtensa.info) }
+  .xt.insn 0 :
+  {
+    KEEP (*(.xt.insn))
+    KEEP (*(.gnu.linkonce.x.*))
+  }
+  .xt.prop 0 :
+  {
+    KEEP (*(.xt.prop))
+    KEEP (*(.xt.prop.*))
+    KEEP (*(.gnu.linkonce.prop.*))
+  }
+  .xt.lit 0 :
+  {
+    KEEP (*(.xt.lit))
+    KEEP (*(.xt.lit.*))
+    KEEP (*(.gnu.linkonce.p.*))
+  }
+  .debug.xt.callgraph 0 :
+  {
+    KEEP (*(.debug.xt.callgraph .debug.xt.callgraph.* .gnu.linkonce.xt.callgraph.*))
+  }
```

Comment:
Having just recopied this same code too, we really should have an "xt-toolchain-debug.ld" include or something rather than continuing to cut and past stuff that quite frankly I don't have any understanding of at all.